packages/react-ui/src/widgets/TableWidgetUI/TableWidgetUI.js

```diff
@@ -0,0 +1,209 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  makeStyles,
+  TableSortLabel,
+  TablePagination
+} from '@material-ui/core';
+
+const useStyles = makeStyles((theme) => ({
+  tableHead: {
+    backgroundColor: theme.palette.common.white,
+    '& .MuiTableCell-head': {
+      border: 'none'
+    },
+    '& .MuiTableCell-head, & .MuiTableCell-head span': {
+      ...theme.typography.caption,
+      color: theme.palette.text.secondary
+    }
+  },
+  tableRow: {
+    maxHeight: theme.spacing(6.5),
+    transition: 'background-color 0.25s ease',
+    '&.MuiTableRow-hover:hover': {
+      cursor: 'pointer',
+      backgroundColor: theme.palette.background.default
+    }
+  },
+  tableCell: {
+    overflow: 'hidden',
+    '& p': {
+      maxWidth: '100%',
+      whiteSpace: 'nowrap',
+      overflow: 'hidden',
+      textOverflow: 'ellipsis'
+    }
+  },
+  pagination: {
+    '& .MuiTablePagination-caption': {
+      ...theme.typography.caption
+    },
+    '& .MuiTablePagination-caption:first-of-type': {
+      color: theme.palette.text.secondary
+    },
+    '& .MuiTablePagination-input': {
+      minHeight: theme.spacing(4.5),
+      border: `2px solid ${theme.palette.divider}`,
+      borderRadius: theme.spacing(0.5)
+    }
+  }
+}));
+
+function TableWidgetUI({
+  columns,
+  rows,
+  sorting,
+  sortBy,
+  sortDirection,
+  onSetSortBy,
+  onSetSortDirection,
+  pagination,
+  totalCount,
+  page,
+  onSetPage,
+  rowsPerPage,
+  rowsPerPageOptions,
+  onSetRowsPerPage,
+  onRowClick
+}) {
+  const classes = useStyles();
+
+  const handleSort = (sortField) => {
+    const isAsc = sortBy === sortField && sortDirection === 'asc';
+    onSetSortDirection(isAsc ? 'desc' : 'asc');
+    onSetSortBy(sortField);
+  };
+
+  const handleChangePage = (_event, newPage) => {
+    onSetPage(newPage + 1);
+  };
+
+  const handleChangeRowsPerPage = (event) => {
+    onSetRowsPerPage(parseInt(event.target.value, 10));
+    onSetPage(1);
+  };
+
+  return (
+    <>
+      <TableContainer>
+        <Table>
+          <TableHeaderComponent
+            columns={columns}
+            sorting={sorting}
+            sortBy={sortBy}
+            sortDirection={sortDirection}
+            onSort={handleSort}
+          />
+          <TableBodyComponent columns={columns} rows={rows} onRowClick={onRowClick} />
+        </Table>
+      </TableContainer>
+      {pagination && (
+        <TablePagination
+          className={classes.pagination}
+          rowsPerPageOptions={rowsPerPageOptions}
+          component='div'
+          count={totalCount}
+          rowsPerPage={rowsPerPage}
+          page={page - 1}
```

Comment:
We need to make this adjustment because `TablePagination` from MUI is zero-based whereas our pagination API is 1-based (see https://github.com/CartoDB/carto-react/blob/master/packages/react-workers/src/workers/features.worker.js#L166)

Or maybe we could change our own implementation to start at page 0 too? Would it be possible @Clebal?

Reply:
@bbecquet totally agree, it's a mistake that page is 1-based. Sorry for my late response, I didn't saw this. Can you open a PR with that change?